pkg/api/api.go

```diff
@@ -165,5 +178,28 @@ func ConfigureAPI(treeID uint) {
 }
 
 func StopAPI() {
+	if api.newEntryPublisher != nil {
+		api.newEntryPublisher.Close()
+	}
 	api.checkpointPublishCancel()
 }
+
+func selectEntryPublisher(ctx context.Context) (pubsub.Publisher, error) {
```

Comment:
I'd like to see this moved into `pkg/pubsub` and mimic a factory pattern, similar to how https://github.com/sigstore/sigstore/blob/main/pkg/signature/kms/kms.go does this for KMS providers.

Reply:
Ack, done.